clients/src/main/java/org/apache/kafka/clients/consumer/KafkaConsumer.java

```diff
@@ -1633,4 +1643,21 @@ private void release() {
         if (refcount.decrementAndGet() == 0)
             currentThread.set(NO_CURRENT_THREAD);
     }
+
+    private class KafkaConsumerMetrics {
+        public final Sensor pollTime;
+
+        public KafkaConsumerMetrics(String metricGrpPrefix) {
+            this.pollTime = metrics.sensor("poll-time");
+            this.pollTime.add(metrics.metricName("poll-time-avg", metricGrpPrefix + "-metrics", "The average time taken for a poll"), new Avg());
+            this.pollTime.add(metrics.metricName("poll-time-max", metricGrpPrefix + "-metrics", "The max time taken for a poll"), new Max());
+            Measurable timeSinceLastPoll = new Measurable() {
+                @Override
+                public double measure(MetricConfig config, long now) {
+                    return now - lastPollEndTimestamp;
+                }
+            };
+            metrics.addMetric(metrics.metricName("time-since-last-poll-ms", metricGrpPrefix + "-metrics", "the number of milliseconds since the last poll completed"), timeSinceLastPoll);
```

Comment:
Instead of showing the time-since-last-poll, should we have the max-time-since-last-poll and average-time-since-last-poll? These two metrics are more informative and stable than the time-since-last-poll since they are measured over a time window.